### PR TITLE
Adding device info to all branch request

### PR DESCRIPTION
**Pull request by sojanpr**
Change : All Branch server requests are added with device info parameters 

Impact : will add additional device param to al requests

Testing :Tested with Testbed app in debug mode 

@aaustin @Sarkar @dmitrig01

### CODE REVIEW

File: Branch-SDK/src/io/branch/referral/DeviceInfo.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        if (thisInstance_ == null) {
+            thisInstance_ = new DeviceInfo(isExternalDebug, sysObserver);
+        }
+        return thisInstance_;
+    }
+
+    private DeviceInfo(boolean isExternalDebug, SystemObserver sysObserver) {
+        hardwareID_ = sysObserver.getUniqueID(isExternalDebug);
+        isHardWareIDReal_ = sysObserver.hasRealHardwareId();
+        careerName_ = sysObserver.getCarrier();
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/referral/DeviceInfo.java]: *carrier

File: Branch-SDK/src/io/branch/referral/DeviceInfo.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    public static DeviceInfo getInstance(boolean isExternalDebug, SystemObserver sysObserver) {
+        if (thisInstance_ == null) {
+            thisInstance_ = new DeviceInfo(isExternalDebug, sysObserver);
+        }
+        return thisInstance_;
+    }
+
+    private DeviceInfo(boolean isExternalDebug, SystemObserver sysObserver) {
+        hardwareID_ = sysObserver.getUniqueID(isExternalDebug);
+        isHardWareIDReal_ = sysObserver.hasRealHardwareId();
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/referral/DeviceInfo.java]: lower case `W`

**Comment by aaustin** [comment]
Looks good to me other than couple minor syntax things :+1:

**Comment by Sarkar** [comment]
solid :+1:

**Pull request merged**